codegenerator/cli/npm/envio/src/db/Table.res

```diff
@@ -166,6 +178,92 @@ let getUnfilteredCompositeIndicesUnsafe = (table): array<array<string>> => {
   )
 }
 
+type sqlParams<'entity> = {
+  dbSchema: S.t<'entity>,
+  quotedFieldNames: array<string>,
+  quotedNonPrimaryFieldNames: array<string>,
+  arrayFieldTypes: array<string>,
+  hasArrayField: bool,
+}
+
+let toSqlParams = (table: table, ~schema) => {
+  let quotedFieldNames = []
+  let quotedNonPrimaryFieldNames = []
+  let arrayFieldTypes = []
+  let hasArrayField = ref(false)
+
+  let dbSchema: S.t<Js.Dict.t<unknown>> = S.schema(s =>
+    switch schema->S.classify {
+    | Object({items}) =>
+      let dict = Js.Dict.empty()
+      items->Belt.Array.forEach(({location, inlinedLocation, schema}) => {
+        let rec coerceSchema = schema =>
+          switch schema->S.classify {
+          | BigInt => BigInt.schema->S.toUnknown
+          | Option(child)
+          | Null(child) =>
+            S.null(child->coerceSchema)->S.toUnknown
+          | Array(child) => {
+              hasArrayField := true
+              S.array(child->coerceSchema)->S.toUnknown
+            }
+          | JSON(_) => {
+              hasArrayField := true
+              schema
+            }
+          | Bool =>
+            // Workaround for https://github.com/porsager/postgres/issues/471
+            S.union([
+              S.literal("t")->S.to(_ => true),
+              S.literal("f")->S.to(_ => false),
+            ])->S.toUnknown
+          | _ => schema
+          }
+
+        let field = switch table->getFieldByDbName(location) {
+        | Some(field) => field
+        | None => raise(NonExistingTableField(location))
+        }
+
+        quotedFieldNames
+        ->Js.Array2.push(inlinedLocation)
+        ->ignore
+        switch field {
+        | Field({isPrimaryKey: false}) =>
+          quotedNonPrimaryFieldNames
+          ->Js.Array2.push(inlinedLocation)
+          ->ignore
+        | _ => ()
+        }
+
+        arrayFieldTypes
+        ->Js.Array2.push(
+          switch field {
+          | Field(f) =>
+            switch f.fieldType {
+            | Custom(fieldType) => `${(Text :> string)}[]::${(fieldType :> string)}`
```

Comment:
I don't really understand what's happening here with Custom type any chance you can add a note?

Reply:
We cast a text type to enum type, since postgres can't do it automatically.